providers/kit/metadata_test.go

```diff
@@ -0,0 +1,224 @@
+	tagsresponse := testutils.DataFromFile(t, "tags.json")
+	webhooksresponse := testutils.DataFromFile(t, "webhooks.json")
+
+	tests := []testroutines.Metadata{
+		{
+			Name:         "Object must be included",
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingObjects},
+		},
+		{
+			Name:  "Successfully describe multiple object with metadata",
+			Input: []string{"broadcasts", "custom_fields", "forms", "subscribers", "tags", "email_templates", "purchases", "segments", "sequences", "webhooks"},
+			Server: mockserver.Switch{
+				Setup: mockserver.ContentJSON(),
+				Cases: []mockserver.Case{{
+					If:   mockcond.PathSuffix("/v4/broadcasts"),
+					Then: mockserver.Response(http.StatusOK, broadcastsresponse),
+				}, {
+					If:   mockcond.PathSuffix("/v4/custom_fields"),
+					Then: mockserver.Response(http.StatusOK, Customfieldsresponse),
+				}, {
+					If:   mockcond.PathSuffix("/v4/email_templates"),
+					Then: mockserver.Response(http.StatusOK, emailtemplatesresponse),
+				}, {
+					If:   mockcond.PathSuffix("/v4/forms"),
+					Then: mockserver.Response(http.StatusOK, formsresponse),
+				}, {
+					If:   mockcond.PathSuffix("/v4/purchases"),
+					Then: mockserver.Response(http.StatusOK, purchasesresponse),
+				}, {
+					If:   mockcond.PathSuffix("/v4/sequences"),
+					Then: mockserver.Response(http.StatusOK, sequencesresponse),
+				}, {
+					If:   mockcond.PathSuffix("/v4/subscribers"),
+					Then: mockserver.Response(http.StatusOK, subscribersresponse),
+				}, {
+					If:   mockcond.PathSuffix("/v4/segments"),
+					Then: mockserver.Response(http.StatusOK, segmentsresponse),
+				}, {
+					If:   mockcond.PathSuffix("/v4/tags"),
+					Then: mockserver.Response(http.StatusOK, tagsresponse),
+				}, {
+					If:   mockcond.PathSuffix("/v4/webhooks"),
+					Then: mockserver.Response(http.StatusOK, webhooksresponse),
+				}},
+			}.Server(),
+			Comparator: func(baseURL string, actual, expected *common.ListObjectMetadataResult) bool {
+				return mockutils.MetadataResultComparator.SubsetFields(actual, expected)
+			},
+			Expected: &common.ListObjectMetadataResult{
+				Result: map[string]common.ObjectMetadata{
+					"broadcasts": {
+						DisplayName: "broadcasts",
+						FieldsMap: map[string]string{
+							"content":           "content",
+							"created_at":        "created_at",
+							"description":       "description",
+							"email_address":     "email_address",
+							"email_template":    "email_template",
+							"id":                "id",
+							"preview_text":      "preview_text",
+							"public":            "public",
+							"published_at":      "published_at",
+							"send_at":           "send_at",
+							"subject":           "subject",
+							"subscriber_filter": "subscriber_filter",
+							"thumbnail_alt":     "thumbnail_alt",
+							"thumbnail_url":     "thumbnail_url",
+						},
+					},
+					"custom_fields": {
+						DisplayName: "custom_fields",
```

Comment:
The display name should be:
`Custom Fields`
The above broadcasts would be:
`Broadcasts`.

I don't see the usage of OpenAPI tool, where `api3.WithDisplayNamePostProcessors` option would help with formatting display names, hence I am thinking you created static script yourself. There are some connectors that are like that, could you then edit `schema.json` to have display names as described? 

These display names are what clients will be seeing in UI.